hummingbird/ml/operator_converters/_tree_commons.py

```diff
@@ -169,20 +169,42 @@ def get_tree_params_and_type(tree_infos, get_tree_parameters, extra_config):
 def get_parameters_for_sklearn_common(tree_infos):
     """
     Parse sklearn-based trees, including
-    SklearnRandomForestClassifier/Regressor and SklearnGradientBoostingClassifier
+    SklearnRandomForestClassifier/Regressor, SklearnGradientBoostingClassifier, and SklearnHistGradientBoostingClassifier
 
     Args:
-        tree_infos: The information representaing a tree (ensemble)
+        tree_infos: The information representing a tree (ensemble)
 
     Returns: The tree parameters wrapped into an instance of
              `operator_converters._tree_commons_TreeParameters`
     """
     trees = tree_infos
-    lefts = trees.tree_.children_left
-    rights = trees.tree_.children_right
-    features = trees.tree_.feature
-    thresholds = trees.tree_.threshold
-    values = trees.tree_.value
+    if hasattr(trees, "nodes"):
+        # SklearnHistGradientBoostingClassifier
+
+        lefts = [trees.nodes[x]["left"] for x in range(len(trees.nodes))]
+        # Use -1 instead of 0 for nodes with no lefts
+        lefts = [idx if idx != 0 else -1 for idx in lefts]
+
+        rights = [trees.nodes[x]["right"] for x in range(len(trees.nodes))]
+        # Use -1 instead of 0 for nodes with no rights
+        rights = [idx if idx != 0 else -1 for idx in rights]
+
+        features = [trees.nodes[x]["feature_idx"] for x in range(len(trees.nodes))]
+        thresholds = [trees.nodes[x]["threshold"] for x in range(len(trees.nodes))]
+
+        values = [trees.nodes[x]["value"] for x in range(len(trees.nodes))]
+        # Convert to numpy.ndarray with shape (s, 1, 1)
+        values = np.array(values)
+        values = values.reshape((values.shape[0], 1, 1))
```

Comment:
`values` has a shape of `(x, 1, 1)` in case of HistGradientBoosting.
Doing `values.reshape(-1, 1)` we will have a shape of `(x, 1)` instead of `(x, 1, 1)`. If this won't cause any problems I can modify it.

Reply:
Yes, can you please put them as `-1, 1`?